homeassistant/components/elmax/config_flow.py

```diff
@@ -0,0 +1,251 @@
+        try:
+            await self._client.get_panel_status(
+                control_panel_id=panel_id, pin=panel_pin
+            )
+            return self.async_create_entry(
+                title=f"Elmax {panel_name}",
+                data={
+                    CONF_ELMAX_PANEL_ID: panel_id,
+                    CONF_ELMAX_PANEL_PIN: panel_pin,
+                    CONF_ELMAX_USERNAME: self._username,
+                    CONF_ELMAX_PASSWORD: self._password,
+                },
+            )
+        except ElmaxBadPinError:
+            errors["base"] = "invalid_pin"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Error occurred")
+            errors["base"] = "unknown_error"
+
+        return self.async_show_form(
+            step_id="panels", data_schema=self._panels_schema, errors=errors
+        )
+
+    async def async_step_reauth(self, user_input=None):
+        """Perform reauth upon an API authentication error."""
+        self._reauth_username = user_input.get(CONF_ELMAX_USERNAME)
+        self._reauth_panelid = user_input.get(CONF_ELMAX_PANEL_ID)
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(self, user_input=None):
+        """Handle reauthorization flow."""
+        errors = {}
+        if user_input is not None:
+            panel_pin = user_input.get(CONF_ELMAX_PANEL_PIN)
+            password = user_input.get(CONF_ELMAX_PASSWORD)
+            entry = await self.async_set_unique_id(self._reauth_panelid)
+
+            # Handle authentication, make sure the panel we are re-authenticating against is listed among results
+            # and verify its pin is correct.
+            try:
+                # Test login.
+                client = Elmax(username=self._reauth_username, password=password)
+                await client.login()
+
+                # Make sure the panel we are authenticating to is still available.
+                panels = await client.list_control_panels()
+                if (
+                    next(filter(lambda p: p.hash == self._reauth_panelid, panels), None)
+                    is None
+                ):
+                    raise NoOnlinePanelsError()
+
+                # Verify the pin is still valid.from
+                await client.get_panel_status(
+                    control_panel_id=self._reauth_panelid, pin=panel_pin
+                )
+
+                # If it is, proceed with configuration update.
+                self.hass.config_entries.async_update_entry(
+                    entry,
+                    data={
+                        CONF_ELMAX_PANEL_ID: self._reauth_panelid,
+                        CONF_ELMAX_PANEL_PIN: panel_pin,
+                        CONF_ELMAX_USERNAME: self._reauth_username,
+                        CONF_ELMAX_PASSWORD: password,
+                    },
+                )
+                await self.hass.config_entries.async_reload(entry.entry_id)
```

Comment:
Since you're updating the config entry one line above consider using https://developers.home-assistant.io/docs/config_entries_options_flow_handler#signal-updates

This might be especially useful if you intend to add an options flow later on

Reply:
I appreciate the suggestion. For now we don't plan to rely on any options flow. In case that changes, I'll surely reconsider it!